threedi_api_qgis_client/widgets/wizard.py

```diff
@@ -394,6 +395,32 @@ def get_laterals_data(self, timesteps_in_seconds=False):
             val["values"] = [[t * seconds_per_unit, v] for (t, v) in val["values"]]
         return laterals_data
 
+    def handle_laterals_header(self, laterals_list, laterals_type, log_error=True):
+        """Fetching first lateral row and handling potential header."""
+        error_message = None
+        if not laterals_list:
+            error_message = "Laterals list is empty!"
+            if log_error is True:
+                self.parent_page.parent_wizard.parent_dock.communication.show_warn(error_message)
+            return error_message
+        header = laterals_list[0]
+        if laterals_type == "1D":
+            if len(header) != 3:
+                error_message = "Wrong timeseries format for 1D laterals!"
+        else:
+            if len(header) != 5:
+                error_message = "Wrong timeseries format for 2D laterals!"
+        if error_message is None:
```

Comment:
Such an `if error_message is None` isn't needed in case you'd raise a custom error. So perhaps it could make the flow a bit clearer?

Reply:
There are already tickets reffering to the errors handling. I would address this issue during work of those tickets and prepare complete solution based on custom errors. But till that time I would stick with current logic.